client/src/components/Profile/MyCourses.js

```diff
@@ -113,7 +114,32 @@ const MyCourses = () => {
         } catch (err) {
             console.log(err);
         }
-    };
+    }
+
+    const handleRemove = async (id) => {
+        //if course has not been added to user 
+        if (!userCourses.some(course => course._id === id)) {
+            setMyCourses(myCourses.filter(c => c._id !== id));
+        } else {
+            try {
+                let groupIds;
+                if (groups && groups.length) {
+                    // filter user groups selecting only the group id's 
+                    // eslint-disable-next-line no-sequences
```

Comment:
Always appreciate some comments to help me understand what you're doing where and why.